tests/e2e/actions.go

```diff
@@ -250,6 +258,121 @@ func (tr Chain) submitTextProposal(
 	tr.waitBlocks(action.Chain, 1, 10*time.Second)
 }
 
+type UpdateConsumerChainAction struct {
+	Chain               ChainID
+	From                ValidatorID
+	ConsumerChain       ChainID
+	NewOwner            string
+	SpawnTime           uint
+	InitialHeight       clienttypes.Height
+	DistributionChannel string
+	TopN                uint32
+	ValidatorsPowerCap  uint32
+	ValidatorSetCap     uint32
+	Allowlist           []string
+	Denylist            []string
+	MinStake            uint64
+	AllowInactiveVals   bool
+}
+
+func (tr Chain) updateConsumerChain(action UpdateConsumerChainAction, verbose bool) {
+
+	spawnTime := tr.testConfig.containerConfig.Now.Add(time.Duration(action.SpawnTime) * time.Millisecond)
+	params := ccvtypes.DefaultParams()
+	initParams := types.ConsumerInitializationParameters{
+		InitialHeight: action.InitialHeight,
+		GenesisHash:   []byte("gen_hash"),
+		BinaryHash:    []byte("bin_hash"),
+		SpawnTime:     spawnTime,
+
+		UnbondingPeriod:                   params.UnbondingPeriod,
+		CcvTimeoutPeriod:                  params.CcvTimeoutPeriod,
+		TransferTimeoutPeriod:             params.TransferTimeoutPeriod,
+		ConsumerRedistributionFraction:    params.ConsumerRedistributionFraction,
+		BlocksPerDistributionTransmission: params.BlocksPerDistributionTransmission,
+		HistoricalEntries:                 params.HistoricalEntries,
+		DistributionTransmissionChannel:   action.DistributionChannel,
+	}
+
+	powerShapingParams := types.PowerShapingParameters{
+		Top_N:              action.TopN,
+		ValidatorsPowerCap: action.ValidatorsPowerCap,
+		ValidatorSetCap:    action.ValidatorSetCap,
+		Allowlist:          action.Allowlist,
+		Denylist:           action.Denylist,
+		MinStake:           action.MinStake,
+		AllowInactiveVals:  action.AllowInactiveVals,
+	}
+
+	consumerId := tr.testConfig.chainConfigs[action.ConsumerChain].ConsumerId
+	update := types.MsgUpdateConsumer{
+		ConsumerId:               string(consumerId),
+		NewOwnerAddress:          action.NewOwner,
+		InitializationParameters: &initParams,
+		PowerShapingParameters:   &powerShapingParams,
+	}
+	tr.UpdateConsumer(action.Chain, action.From, update)
+}
+
+type CreateConsumerChainAction struct {
+	Chain               ChainID
+	From                ValidatorID
+	ConsumerChain       ChainID
+	InitialHeight       clienttypes.Height
+	SpawnTime           uint
+	DistributionChannel string
+	TopN                uint32
+	ValidatorsPowerCap  uint32
+	ValidatorSetCap     uint32
+	Allowlist           []string
+	Denylist            []string
+	MinStake            uint64
+	AllowInactiveVals   bool
+}
+
+// createConsumerChain creates and initializes a consumer chain
+func (tr Chain) createConsumerChain(action CreateConsumerChainAction, verbose bool) {
+
+	spawn_time := tr.testConfig.containerConfig.Now.Add(time.Duration(action.SpawnTime) * time.Millisecond)
```

Comment:
Can we use camel case throughout? In later parts as well (e.g., you use `consumer_id`).